### ISSUE TITLE
Poor tracking quality on some Qualcomm-based Samsung Galaxy S9 devices

### ISSUE THREAD

**Issue by NWalker1208** [google-ar/arcore-android-sdk#419]
My Samsung Galaxy S9 has the model number SM-G960U, running Android 8.0.0. I am experiencing bugs with ARCore that prevent me from being able to use ARCore-based apps. I have tested my phone on three such apps, two of which I know worked my last device, an S7 Edge running Android Nougat, and the last of which I have only tested on my current device. I currently have ARCore version 1.2.180425134 installed.

Just a Line by Google (version 2.0.0)
Drawings fly around wildly and unpredictably.

ARuler by Grymala (version 1.1.2)
Cannot detect surfaces to measure, even in optimal conditions.

IKEA Place by IKEA (version 2.2.86)
When furniture is selected, it rarely appears, and when it does, it appears distant and jumps around the view when the phone is moved.

All of these seem to be related to the phone's ability to detect it's position in space. If anyone else is experiencing this issue, please speak up. I believe the Galaxy S9 is supposed to have ARCore support at this point, but the implementation is completely broken for me.

If you would like me to test any other apps, or if any more information is needed, please let me know. If this is not the write place to mention this issue, please let me know where I should ask for support on this issue.

**Comment by inio** [comment]
It sounds like you have a device that's at the extreme outliers in terms of calibration.  There's not much we can do.

If possible, what is the build fingerprint of this device? (`adb shell getprop ro.build.fingerprint`)

**Comment by NWalker1208** [comment]
Build fingerprint is:
samsung/starqltesq/starqltesq:8.0.0/R16NW/G960USQU1ARBG:user/release-keys

Are you sure that there is nothing that can be done? The fact that this is so badly broken on my device makes me concerned that others will likely have the same problem. It's improbable that this is a rare situation. I want others with a Galaxy S9 to state their experience with ARCore so we can tell if it is a common issue or not. If this is a real issue, could better calibration be done? Could I calibrate my own device? I really want to use ARCore, so what could I do, or who could I seek help from to resolve this issue?

**Comment by NWalker1208** [comment]
Also are you saying my specific, individual device is at fault, or that Galaxy S9's in general are outliers in terms of calibration?

**Comment by NWalker1208** [comment]
By the way, just for reference for what is occurring, see this video: [Video on Google Photos](https://photos.app.goo.gl/Jabhhmd1ACe2jxG49)

It seems to half-work, but drifts wildly, despite the phone barely moving at all. I have not studied the inner workings of ARCore, but I image it uses some data from the gyro and accelerometers, so why, when the phone isn't moving and the camera view isn't changing, would it perceive such wild movement?

**Comment by inio** [comment]
Given that we see ARCore working well on the majority of S9 devices, I'd say that yes, your specific device is particular far from the average calibration we include in arcore.

For best performance, I suggest translating (moving) the device substantially before you try to interact with AR.  Without parallax, we can't build a 3D map and scale is largely unconstrained.

**Comment by nuudles** [comment]
I am also experiencing similar issues with my Galaxy S9 using ARCore. My Nexus 6p works well with the AR apps I have tried. My build fingerprint is:

`samsung/starqltesq/starqltesq:8.0.0/R16NW/G960USQU1ARB7:user/release-keys`

**Comment by chadcspencer** [comment]
I to am experiencing the problem. I have found other sites that have acknowledged that it is a issue with a samsung driver. 
See: 
http://piunikaweb.com/2018/04/24/galaxy-s9-arcore-support-delayed-due-to-bug-at-samsungs-end/

**Comment by inio** [comment]
The driver issue was worked around in ARCore 1.2.  This issue seems to be accelerometer-related.

I showed our tracking group that video and they feel this is a proper bug.  Reopening.

**Comment by inio** [comment]
@chadcspencer Do you have a "starqltesq" as well?

**Comment by inio** [comment]
Small update: We've identified from analytics that there is a tracking quality issue affecting some S9 and S9+ devices.  We believe some changes in the upcoming 1.3 release will improve things, but are not yet sure how much.

**Comment by srschmidt** [comment]
I'm experiencing the same issue on my S9+ device. Happy to share additional details if needed. 

Build fingerprint: 
`samsung/star2qltesq/star2qltesq:8.0.0/R16NW/G965USQU1ARB6:user/release-keys`

**Comment by chadcspencer** [comment]
I hope 1.3 fixes it because it is completely unusable in it's current state. I know other AR apps built in work so I it can't be a sensor issue. It has to be driver related.

**Comment by inio** [comment]
@chadcspencer We're crossing our fingers.

We're still trying to identify a device that demonstrates this behavior as dramatically as is shown.

**Comment by NWalker1208** [comment]
@inio I've found that the issue occurs most of the time, but in certain situations the tracking is much better. I will try to keep track of where it works.

**Comment by jruppert83** [comment]
I am also having this issue on the Galaxy S9 SM-G960U, build R16NW.G960USQU2ARC6

**Comment by inio** [comment]
So @NWalker1208 @nuudles @chadcspencer and @srschmidt: We just release ARCore 1.3.  Please let me know if this improves the behavior of ARCore on your devices.

**Comment by chadcspencer** [comment]
I see no marked improvements at all with 1.3

*Thanks you,*

Chad Spencer
*chadcspencer@gmail.com <chadcspencer@gmail.com>*

**Comment by slukow1111** [comment]
I am having exactly the same issue on my Verizon Galaxy S9+. None of the above mentioned AR Core based apps are usable.

**Comment by NWalker1208** [comment]
@inio Any updates on this issue?

**Comment by chadcspencer** [comment]
@inio I appreciate the effort to fix it.

**Comment by ricsmania** [comment]
I have a Brazilian Galaxy S9+, model SM-G9650 with Snapdragon 845 and I have the same issues. My device is marked as incompatible, but I have tried sideloading the 1.3 ARCore APK as well as some apps, and the behavior is very erratic, very similar to the video @NWalker1208 posted.

The build fingerprint is:

`samsung/star2qltezh/star2qltechn:8.0.0/R16NW/G9650ZHU2ARE6:user/release-keys`

Coincidentally I also had a S7 edge before, and it worked perfectly with ARCore.

**Comment by TheAwesomeGcdm** [comment]
Not working on latest build of ARcore on SM-G960U, but working completely fine on my old Galaxy S7.

**Comment by nuudles** [comment]
@inio I finally had a chance to test with the latest version of ARCore and it looks like the tracking is very much improved on my Galaxy S9. Now the performance is about the same as with my other devices.

**Comment by chadcspencer** [comment]
It is for the first time actually usable. Although coming form my S8 I would say it's still not as smooth.

**Comment by slukow1111** [comment]
There has been no change in performance for me. It is still non-existent. No ARCore based apps are usable still.

**Comment by inio** [comment]
@chadcspencer There's been no change since the initial release of 1.3.  Did you get an android update recently?

@slukow1111 @TheAwesomeGcdm @ricsmania @chadcspencer @srschmidt (and anyone else affected) We're sorry 1.3 doesn't seem to have helped this much if at all.  It would like to help us track this down, please drop me an email at *`[my GitHub username]`*`@google.com` with the answers to these questions:

* How badly does ARCore work on your device (does it *ever* start tracking?  for how long?)?
* What version of the ARCore APK is installed (including the extra digits at the end)?
* What is your device's build fingerprint (`adb shell getprop ro.build.fingerprint`)?
* What is your device's serial number (`adb shell getprop ro.serialno`)?
* What is the issuing carrier of any SIM card(s) in the device?

**Comment by Studiociteroni** [comment]
I've the same problem with an Samsung S8. I resolved uninstalling ARCore 1.3.0 app and reinstalling the old ARCore 1.1.0. Now the surfaces tracking works well, however I hope these problems are fixed with the next ARCore release.

**Comment by chadcspencer** [comment]
Found version 1.3.180604066 on APKMIRROR, and it works very good on my S9. This is the first build usable on my S9! Great work.

**Comment by inio** [comment]
@Studiociteroni I'm going to lump that into #478

**Comment by ricsmania** [comment]
I have version 1.3.180604066 and it doesn't work for me. I also tried downgrading to 1.1 but nothing changed.

**Comment by inio** [comment]
All Affected Users:  If you could capture a full logcat of a 15-30 second session where ARCore is misbehaving, this might help us track down the cause of the failure.  Since this thread is already quite long, please send me the logcat (*`[my GitHub username]`*`@google.com`) or put it on a pastebin or similar and reply with the link.  Bonus points if you can capture a video at the same time as the logcat and share that as well (but the logcat is much more important).

**Comment by slukow1111** [comment]
Sent.

**Comment by michaelheiml** [comment]
Hi, I've experienced similar issues on my Galaxy S7 Edge (with original firmware and also Custom ROM Havoc 8.1) - no tracking was possible at all, only found points but no planes ever appeared in every app I tested. I was able to solve it by recalibrating the accelerator sensor and rebooting once. The app I used is called "Accelerometer Calibration". @inio if that solves the issue for the other users as well, I suggest to add some sort of calibration mode to ARCore?

**Comment by jruppert83** [comment]
Hi Ian,

I just figured out something that might be part of the problem.  After
installing the ARCore, Samsung's Gear VR service went crazy on battery
usage.  I disabled the Gear VR service and now the ARCore seems functional.

Regards,

John

**Comment by inio** [comment]
Thank you everyone for the logs and videos.  Unfortunately, none of them have turned up a conclusive reason why ARCore is not working.  We're continuing to investigate this and working with both Samsung and Qualcomm to find the root cause.

**Comment by chadcspencer** [comment]
I can confirm that if I launch "Just a line" directly after reboot. It definitely  is tracking correctly. So this is good news inio. Thanks for the update.

**Comment by ricsmania** [comment]
@inio I also tested Just a Line after rebooting. It was erratic for the first 5 seconds or so, but after that it started working fine. Thanks for the update!

**Comment by rplevy** [comment]
I am experiencing the exact same problems on my brand new Galaxy S9.

**Comment by matinas** [comment]
I've just bought a Galaxy S9 SM-G960F/DS with Snapdragon 845 and I'm having this very same issue with any ARCore app. Does the update that solves it already rolled out? Otherwise, do we have to reset the device every time before running an ARCore app in the meantime? Will try that workaround asap and check whether it solves the poor tracking. Thanks!

**Comment by Organologic** [comment]
Hi everyone!

I believe that this issue are related with battery saving system control. I've been facing the same problem and the thing is that when I try to use any ARCore App perfectly, I have to restart the phone without battery saving on. It works for a few minutes.

I hope that Samsung Developers find a solution soon. Thanks!

**Comment by matinas** [comment]
Hi, I'm starting developing for ARCore and because of this issue I'm restarting my S9 way lot more than I expected (something between 10-20 times per hour to debug and stuff). Does someone know if restarting the phone so many times can somehow harm it in the long term? Thanks!

**Comment by jpetro416** [comment]
Having similar issues on the Note 9, blue dots appear occasionally but no surface tracking or anything... it sucks because I specifically got the Note 9 for AR dev (plus all the goodies that come with a Note). Any solution? All it does it show tiny blue dots every once in a while...no plane detection though.

**Comment by inio** [comment]
Official Update:  We've identified a possible root cause of this issue and are working with Samsung to release a fix for consumers. In the interim to fix tracking issues and unblock your development, please disable iris unlock on your phone (use some other unlock mechanism). If that does not work, please also reboot your phone.

**Comment by matinas** [comment]
Can confirm that the tracking is working fine after disabling the iris scanning feature. Will be waiting for the official fix but I must say this workaround is way better than restarting the phone every time. Thanks, @inio!

**Comment by rplevy** [comment]
On my Samsung S9 I disabled Iris Unlock and it still has the problem that tracking surfaces only working for a little while after booting up, and then stops working.

**Comment by inio** [comment]
We were able to work with Samsung to find a workaround and this should be fixed with [ARCore 1.5](https://github.com/google-ar/arcore-android-sdk/releases/tag/v1.5.0) which was released today.  Go ahead and turn Iris Unlock back on and let us know how things go.

**Comment by Shifter2600** [comment]
Confirmed it works now on my Samsung s9+. Thx

**Comment by michaelheiml** [comment]
Although not mentioned, I tested ARCore 1.5 on Galaxy S7 Edge (SM-G935F) and it does not work. The plane detection seems to work faster and it finds planes repeatedly within 2-4 sec, but they are lost within a few seconds (before 1.5, planes were found once after device restart, then never again). Sometimes it looks as the planes found "swim away" out of screen, then tracking is lost. Sometimes the tracking is lost immediately, planes disappear instantly. Tested with the AR Measure app from Google.

**Comment by inio** [comment]
@michaelheiml Please open a separate issue for that.

**Comment by chadcspencer** [comment]
I don't. Not completely sure how to generate. Can you point me to a faq?

**Comment by lorenzoSpacelens** [comment]
Hi, the problem still persist. I have tried all version from 1.29.0 down to 1.20.0 and only the last one works.
We are building an app with augmented images and the plane is tracked in intermittent mode. Anchored video appear, disappear, appear all the time.

We have tried with Xiaomi, Pixel, Huawei and everything work perfectly.

These are the technical info of the Samsung phone

![Samsung_technical_info](https://user-images.githubusercontent.com/89907448/151236140-95954e05-2c08-4068-b6c8-def11b535e25.jpeg)
